bin/node-template/pallets/template/Cargo.toml

```diff
@@ -8,21 +8,22 @@ homepage = "https://substrate.dev"
 repository = "https://github.com/substrate-developer-hub/substrate-node-template/"
 description = "FRAME pallet template for defining custom runtime logic."
 readme = "README.md"
+publish = false
 
 [package.metadata.docs.rs]
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
 codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false, features = ["derive"] }
-frame-support = { default-features = false, version = "3.0.0", path = "../../../../frame/support" }
-frame-system = { default-features = false,  version = "3.0.0",  path = "../../../../frame/system" }
-frame-benchmarking = { default-features = false,  version = "3.1.0",  path = "../../../../frame/benchmarking", optional = true }
+frame-support = { default-features = false, version = "4.0.0-dev", path = "../../../../frame/support" }
+frame-system = { default-features = false,  version = "4.0.0-dev",  path = "../../../../frame/system" }
+frame-benchmarking = { default-features = false,  version = "4.0.0-dev",  path = "../../../../frame/benchmarking", optional = true }
 
 [dev-dependencies]
 serde = { version = "1.0.119" }
-sp-core = { default-features = false, version = "3.0.0", path = "../../../../primitives/core" }
-sp-io = { default-features = false, version = "3.0.0", path = "../../../../primitives/io" }
-sp-runtime = { default-features = false, version = "3.0.0", path = "../../../../primitives/runtime" }
+sp-core = { default-features = false, version = "4.0.0-dev", path = "../../../../primitives/core" }
```

Comment:
does it make sense to omit versions for dev-deps only leaving path as you did in https://github.com/paritytech/parity-common/pull/552#discussion_r661848985?

Reply:
That's something that I have discovered recently and will probably impact the unleash tool as well. As of now, `cargo unleash` automatically removes dev-dependencies to break these cycles but if there is 1st party support for that in Cargo than we should rethink what `cargo unleash` does and probably migrate to `path`-only dev-dependencies in the Substrate repository in the near future.

We talked with @gnunicorn already about that and we will address that somehow but for now I would prefer not to touch that in this PR.